src/briefcase/__main__.py

```diff
@@ -1,43 +1,43 @@
 import sys
 
-from .cmdline import parse_cmdline
-from .console import Log
-from .exceptions import BriefcaseError, HelpText
+from briefcase.cmdline import parse_cmdline
+from briefcase.console import Console, Log
+from briefcase.exceptions import BriefcaseError, HelpText
 
 
 def main():
-    log = Log()
+    result = 0
     command = None
+    logger = Log()
+    console = Console()
     try:
-        command, options = parse_cmdline(sys.argv[1:])
-        # Replace the top-level logger with the one used by the command.
-        # This is needed to preserve extra logging detail.
-        log = command.logger
+        Command, extra_cmdline = parse_cmdline(sys.argv[1:])
+        command = Command(logger=logger, console=console)
+        options = command.parse_options(extra=extra_cmdline)
         command.check_obsolete_data_dir()
         command.parse_config("pyproject.toml")
         command(**options)
-        result = 0
```

Comment:
This refactoring is to promote visibility of the creation of `Log`, `Console`, and `Command`.